src/sim_engine.rs

```diff
@@ -48,12 +48,10 @@ impl Engine for SimEngine {
 
     fn list_pools(&self) -> StratisResult<BTreeMap<String, Box<StratisPool>>> {
 
-        let mut stratis_pools = BTreeMap::new();
-
-        for (name, pool) in &self.pools {
-
-            stratis_pools.insert(name.clone(), pool.copy());
-        }
+        let stratis_pools = self.pools.iter().fold(BTreeMap::new(), |mut acc, x| {
```

Comment:
BTreemap implements FromIterator so maybe see if a chain of iter(), cloned(), maybe a map()?, and then collect() lets you do this and avoid the fold